app/src/main/java/org/stepic/droid/ui/dialogs/ChooseStorageDialog.kt

```diff
@@ -43,16 +46,24 @@ class ChooseStorageDialog: DialogFragment() {
         val youWantMoveDataDialog = WantMoveDataDialog.newInstance()
         youWantMoveDataDialog.setTargetFragment(targetFragment, WantMoveDataDialog.REQUEST_CODE)
 
+        val view = View.inflate(context, R.layout.dialog_listview, null)
+        val selectionItems = view.findViewById<ListView>(R.id.listChoices)
+
+        selectionItems.adapter = ArrayAdapter<String>(requireContext(), R.layout.simple_list_item_single_choice, headers)
+        selectionItems.choiceMode = ListView.CHOICE_MODE_SINGLE
+        selectionItems.setItemChecked(indexChosen, true)
+        selectionItems.setOnItemClickListener { _, _, position, _ ->
+            if (position != indexChosen && !youWantMoveDataDialog.isAdded) {
+                youWantMoveDataDialog.targetLocation = storageOptions[position]
+                youWantMoveDataDialog.show(fragmentManager, null)
+            }
+            dismiss()
+        }
+
         return AlertDialog.Builder(requireContext())
                 .setTitle(R.string.choose_storage_title)
+                .setView(view)
```

Comment:
Why we can't use `public AlertDialog.Builder setSingleChoiceItems(ListAdapter adapter, int checkedItem, OnClickListener listener)` ?